src/main/java/appeng/helpers/patternprovider/PatternProviderLogic.java

```diff
@@ -383,6 +401,7 @@ public void resetCraftingLock() {
 
     private void onPushPatternSuccess(IPatternDetails pattern) {
         resetCraftingLock();
+        tryCancelCraft(pattern);
```

Comment:
Maybe rename this to tryAutoCompleteCraft? from the name alone I got very confused reading through this method